Installation issue: hdf5

### Steps to reproduce the issue

```console
$ spack spec -I <spec>
Input spec
--------------------------------
 -   hdf5+fortran+hl+mpi+szip+threadsafe

Concretized
--------------------------------
 -   hdf5@1.14.3%apple-clang@15.0.0~cxx+fortran+hl~ipo~java~map+mpi+shared~subfiling+szip+threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=darwin-ventura-m1
[e]      ^cmake@3.27.8%apple-clang@15.0.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=darwin-ventura-m1
[+]      ^gmake@4.4.1%apple-clang@15.0.0~guile build_system=generic arch=darwin-ventura-m1
[+]      ^libaec@1.0.6%apple-clang@15.0.0~ipo+shared build_system=cmake build_type=Release generator=make arch=darwin-ventura-m1
[+]      ^openmpi@4.1.6%apple-clang@15.0.0~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~internal-pmix~java~legacylaunchers~lustre~memchecker~openshmem~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=darwin-ventura-m1
[+]          ^gnuconfig@2022-09-17%apple-clang@15.0.0 build_system=generic arch=darwin-ventura-m1
[+]          ^hwloc@2.9.1%apple-clang@15.0.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl~pci~rocm build_system=autotools libs=shared,static arch=darwin-ventura-m1
[+]              ^libxml2@2.10.3%apple-clang@15.0.0+pic~python+shared build_system=autotools arch=darwin-ventura-m1
[+]                  ^libiconv@1.17%apple-clang@15.0.0 build_system=autotools libs=shared,static arch=darwin-ventura-m1
[+]                  ^xz@5.4.1%apple-clang@15.0.0~pic build_system=autotools libs=shared,static arch=darwin-ventura-m1
[+]              ^ncurses@6.4%apple-clang@15.0.0~symlinks+termlib abi=none build_system=autotools arch=darwin-ventura-m1
[e]          ^openssh@9.0p1%apple-clang@15.0.0+gssapi build_system=autotools patches=3505c58,d886b98 arch=darwin-ventura-m1
[+]          ^perl@5.38.0%apple-clang@15.0.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=darwin-ventura-m1
[+]              ^berkeley-db@18.1.40%apple-clang@15.0.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=darwin-ventura-m1
[+]              ^bzip2@1.0.8%apple-clang@15.0.0~debug~pic+shared build_system=generic arch=darwin-ventura-m1
[+]                  ^diffutils@3.9%apple-clang@15.0.0 build_system=autotools arch=darwin-ventura-m1
[+]              ^gdbm@1.23%apple-clang@15.0.0 build_system=autotools arch=darwin-ventura-m1
[+]                  ^readline@8.2%apple-clang@15.0.0 build_system=autotools patches=bbf97f1 arch=darwin-ventura-m1
[+]          ^pmix@5.0.1%apple-clang@15.0.0~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=darwin-ventura-m1
[+]              ^libevent@2.1.12%apple-clang@15.0.0+openssl build_system=autotools arch=darwin-ventura-m1
[e]                  ^autoconf@2.71%apple-clang@15.0.0 build_system=autotools arch=darwin-ventura-m1
[e]                  ^automake@1.16.5%apple-clang@15.0.0 build_system=autotools arch=darwin-ventura-m1
[e]                  ^libtool@2.4.7%apple-clang@15.0.0 build_system=autotools arch=darwin-ventura-m1
[e]                  ^openssl@3.2.0%apple-clang@15.0.0~docs+shared build_system=generic certs=mozilla arch=darwin-ventura-m1
[+]      ^pkgconf@1.9.5%apple-clang@15.0.0 build_system=autotools arch=darwin-ventura-m1
[+]      ^zlib-ng@2.1.4%apple-clang@15.0.0+compat+opt build_system=autotools arch=darwin-ventura-m1
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Installing hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva [16/16]
==> No binary for hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva found: installing from source
==> Using cached archive: /Users/mathomp4/spack-mathomp4/var/spack/cache/_source-cache/archive/09/09cdb287aa7a89148c1638dd20891fdbae08102cf433ef128fd345338aa237c7.tar.gz
==> Ran patch() for hdf5
==> hdf5: Executing phase: 'cmake'
==> hdf5: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j6'

4 errors found in build log:
     340     [  0%] Linking C executable ../../bin/H5match_types
     341     cd /private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-bu
             ild-5jf4yhy/fortran/src && /Users/mathomp4/.homebrew/brew/Cellar/cmake/3.27.8/bin/cmake -E cmake_link_script CMakeFiles/H5match_types.dir/link.txt
              --verbose=1
     342     /Users/mathomp4/spack-mathomp4/lib/spack/env/clang/clang  -std=c99 -Wno-error=implicit-function-declaration -O3 -DNDEBUG -arch arm64 -isysroot /Ap
             plications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=13.0 -Wl,-search_paths_first
             -Wl,-headerpad_max_install_names -Wl,-flat_namespace -Wl,-commons,use_dylibs -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-cl
             ang-15.0.0/hwloc-2.9.1-76ybx4ysxv7vxokyswaj23qx4y7fhfrx/lib -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/libeve
             nt-2.1.12-q3whzyhd3mcanzdgopxz664myodxshyy/lib -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/pmix-5.0.1-siobk427
             4yu5ywcx3z7ajkvh6vjsa43f/lib -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/hwloc-2.9.1-76ybx4ysxv7vxokyswaj23qx4
             y7fhfrx/lib -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/libevent-2.1.12-q3whzyhd3mcanzdgopxz664myodxshyy/lib -
             L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/pmix-5.0.1-siobk4274yu5ywcx3z7ajkvh6vjsa43f/lib  CMakeFiles/H5match
             _types.dir/H5match_types.c.o -o ../../bin/H5match_types  -Wl,-rpath,@executable_path/../lib -Wl,-rpath,@executable_path/ -Wl,-rpath,@loader_path/.
             ./lib -Wl,-rpath,@loader_path/
     343     [  0%] Building C object src/CMakeFiles/hdf5-static.dir/H5build_settings.c.o
     344     cd /private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-bu
             ild-5jf4yhy/src && /Users/mathomp4/spack-mathomp4/lib/spack/env/clang/clang  -I/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-sta
             ge/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src -I/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/
             spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src/H5FDsubfiling -I/private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/ma
             thomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-build-5jf4yhy/src -I/Users/mathomp4/spack-mathomp4/opt/spack/dar
             win-ventura-m1/apple-clang-15.0.0/zlib-ng-2.1.4-x7qn2eenbtlrvogqwblgubufm3af75lh/include -I/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura
             -m1/apple-clang-15.0.0/libaec-1.0.6-2burp52r3trdkw7oyfay33yc7accx7gv/include -isystem /Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/a
             pple-clang-15.0.0/openmpi-4.1.6-5533xhhvzvbsamx55vyh5xvac3hidqgz/include -std=c99 -Wno-error=implicit-function-declaration -O3 -DNDEBUG -std=gnu99
              -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=13.
             0 -fPIC -Wall -Warray-bounds -Wcast-qual -Wconversion -Wdouble-promotion -Wextra -Wformat=2 -Wframe-larger-than=16384 -Wimplicit-fallthrough -Wnul
             l-dereference -Wunused-const-variable -Wwrite-strings -Wpedantic -Wvolatile-register-var -Wno-c++-compat -Wbad-function-cast -Wimplicit-function-d
             eclaration -Wincompatible-pointer-types -Wmissing-declarations -Wpacked -Wshadow -Wswitch -Wno-error=incompatible-pointer-types-discards-qualifier
             s -Wunused-function -Wunused-variable -Wunused-parameter -Wcast-align -Wformat -Wno-missing-noreturn -MD -MT src/CMakeFiles/hdf5-static.dir/H5buil
             d_settings.c.o -MF CMakeFiles/hdf5-static.dir/H5build_settings.c.o.d -o CMakeFiles/hdf5-static.dir/H5build_settings.c.o -c /private/var/folders/31
             /16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-build-5jf4yhy/src/H5build_se
             ttings.c
     345     ld: unknown options: -commons
  >> 346     clang: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 347     make[2]: *** [fortran/src/CMakeFiles/H5match_types.dir/build.make:100: bin/H5match_types] Error 1
     348     make[2]: Leaving directory '/private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvye
             hzyuzte6xzn53wva/spack-build-5jf4yhy'
  >> 349     make[1]: *** [CMakeFiles/Makefile2:1841: fortran/src/CMakeFiles/H5match_types.dir/all] Error 2
     350     make[1]: *** Waiting for unfinished jobs....
     351     [  0%] Building C object src/CMakeFiles/hdf5-static.dir/H5checksum.c.o
     352     cd /private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-bu
             ild-5jf4yhy/src && /Users/mathomp4/spack-mathomp4/lib/spack/env/clang/clang  -I/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-sta
             ge/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src -I/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/
             spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src/H5FDsubfiling -I/private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/ma
             thomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-build-5jf4yhy/src -I/Users/mathomp4/spack-mathomp4/opt/spack/dar
             win-ventura-m1/apple-clang-15.0.0/zlib-ng-2.1.4-x7qn2eenbtlrvogqwblgubufm3af75lh/include -I/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura
             -m1/apple-clang-15.0.0/libaec-1.0.6-2burp52r3trdkw7oyfay33yc7accx7gv/include -isystem /Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/a
             pple-clang-15.0.0/openmpi-4.1.6-5533xhhvzvbsamx55vyh5xvac3hidqgz/include -std=c99 -Wno-error=implicit-function-declaration -O3 -DNDEBUG -std=gnu99
              -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=13.
             0 -fPIC -Wall -Warray-bounds -Wcast-qual -Wconversion -Wdouble-promotion -Wextra -Wformat=2 -Wframe-larger-than=16384 -Wimplicit-fallthrough -Wnul
             l-dereference -Wunused-const-variable -Wwrite-strings -Wpedantic -Wvolatile-register-var -Wno-c++-compat -Wbad-function-cast -Wimplicit-function-d
             eclaration -Wincompatible-pointer-types -Wmissing-declarations -Wpacked -Wshadow -Wswitch -Wno-error=incompatible-pointer-types-discards-qualifier
             s -Wunused-function -Wunused-variable -Wunused-parameter -Wcast-align -Wformat -Wno-missing-noreturn -MD -MT src/CMakeFiles/hdf5-static.dir/H5chec
             ksum.c.o -MF CMakeFiles/hdf5-static.dir/H5checksum.c.o.d -o CMakeFiles/hdf5-static.dir/H5checksum.c.o -c /var/folders/31/16x8_l1s72s2kw0w7byr03qm0
             000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src/H5checksum.c
     353     [  0%] Building C object src/CMakeFiles/hdf5-static.dir/H5dbg.c.o
     354     cd /private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-bu
             ild-5jf4yhy/src && /Users/mathomp4/spack-mathomp4/lib/spack/env/clang/clang  -I/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-sta
             ge/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src -I/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/
             spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src/H5FDsubfiling -I/private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/ma
             thomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-build-5jf4yhy/src -I/Users/mathomp4/spack-mathomp4/opt/spack/dar
             win-ventura-m1/apple-clang-15.0.0/zlib-ng-2.1.4-x7qn2eenbtlrvogqwblgubufm3af75lh/include -I/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura
             -m1/apple-clang-15.0.0/libaec-1.0.6-2burp52r3trdkw7oyfay33yc7accx7gv/include -isystem /Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/a
             pple-clang-15.0.0/openmpi-4.1.6-5533xhhvzvbsamx55vyh5xvac3hidqgz/include -std=c99 -Wno-error=implicit-function-declaration -O3 -DNDEBUG -std=gnu99
              -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=13.
             0 -fPIC -Wall -Warray-bounds -Wcast-qual -Wconversion -Wdouble-promotion -Wextra -Wformat=2 -Wframe-larger-than=16384 -Wimplicit-fallthrough -Wnul
             l-dereference -Wunused-const-variable -Wwrite-strings -Wpedantic -Wvolatile-register-var -Wno-c++-compat -Wbad-function-cast -Wimplicit-function-d
             eclaration -Wincompatible-pointer-types -Wmissing-declarations -Wpacked -Wshadow -Wswitch -Wno-error=incompatible-pointer-types-discards-qualifier
             s -Wunused-function -Wunused-variable -Wunused-parameter -Wcast-align -Wformat -Wno-missing-noreturn -MD -MT src/CMakeFiles/hdf5-static.dir/H5dbg.
             c.o -MF CMakeFiles/hdf5-static.dir/H5dbg.c.o.d -o CMakeFiles/hdf5-static.dir/H5dbg.c.o -c /var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp
             4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-src/src/H5dbg.c
     355     [  0%] Building C object src/CMakeFiles/hdf5-shared.dir/H5checksum.c.o

     ...

     1735    /Users/mathomp4/spack-mathomp4/lib/spack/env/clang/clang  -std=c99 -Wno-error=implicit-function-declaration -O3 -DNDEBUG -arch arm64 -isysroot /Ap
             plications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=13.0 -dynamiclib -Wl,-headerp
             ad_max_install_names -current_version 313.0.0 -compatibility_version 313.0.0 -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-cl
             ang-15.0.0/hwloc-2.9.1-76ybx4ysxv7vxokyswaj23qx4y7fhfrx/lib -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/libeve
             nt-2.1.12-q3whzyhd3mcanzdgopxz664myodxshyy/lib -L/Users/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/pmix-5.0.1-siobk427
             4yu5ywcx3z7ajkvh6vjsa43f/lib -compatibility_version 310.0.0 -current_version 310.3.0 -o ../bin/libhdf5.310.3.0.dylib -install_name @rpath/libhdf5.
             310.dylib "CMakeFiles/hdf5-shared.dir/H5.c.o" "CMakeFiles/hdf5-shared.dir/H5build_settings.c.o" "CMakeFiles/hdf5-shared.dir/H5checksum.c.o" "CMake
             Files/hdf5-shared.dir/H5dbg.c.o" "CMakeFiles/hdf5-shared.dir/H5mpi.c.o" "CMakeFiles/hdf5-shared.dir/H5system.c.o" "CMakeFiles/hdf5-shared.dir/H5ti
             mer.c.o" "CMakeFiles/hdf5-shared.dir/H5trace.c.o" "CMakeFiles/hdf5-shared.dir/H5A.c.o" "CMakeFiles/hdf5-shared.dir/H5Abtree2.c.o" "CMakeFiles/hdf5
             -shared.dir/H5Adense.c.o" "CMakeFiles/hdf5-shared.dir/H5Adeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Aint.c.o" "CMakeFiles/hdf5-shared.dir/H5Atest.c
             .o" "CMakeFiles/hdf5-shared.dir/H5AC.c.o" "CMakeFiles/hdf5-shared.dir/H5ACdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5ACmpio.c.o" "CMakeFiles/hdf5-shar
             ed.dir/H5ACproxy_entry.c.o" "CMakeFiles/hdf5-shared.dir/H5B.c.o" "CMakeFiles/hdf5-shared.dir/H5Bcache.c.o" "CMakeFiles/hdf5-shared.dir/H5Bdbg.c.o"
              "CMakeFiles/hdf5-shared.dir/H5B2.c.o" "CMakeFiles/hdf5-shared.dir/H5B2cache.c.o" "CMakeFiles/hdf5-shared.dir/H5B2dbg.c.o" "CMakeFiles/hdf5-shared
             .dir/H5B2hdr.c.o" "CMakeFiles/hdf5-shared.dir/H5B2int.c.o" "CMakeFiles/hdf5-shared.dir/H5B2internal.c.o" "CMakeFiles/hdf5-shared.dir/H5B2leaf.c.o"
              "CMakeFiles/hdf5-shared.dir/H5B2stat.c.o" "CMakeFiles/hdf5-shared.dir/H5B2test.c.o" "CMakeFiles/hdf5-shared.dir/H5C.c.o" "CMakeFiles/hdf5-shared.
             dir/H5Cdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5Centry.c.o" "CMakeFiles/hdf5-shared.dir/H5Cepoch.c.o" "CMakeFiles/hdf5-shared.dir/H5Cimage.c.o" "CMa
             keFiles/hdf5-shared.dir/H5Cint.c.o" "CMakeFiles/hdf5-shared.dir/H5Clog.c.o" "CMakeFiles/hdf5-shared.dir/H5Clog_json.c.o" "CMakeFiles/hdf5-shared.d
             ir/H5Clog_trace.c.o" "CMakeFiles/hdf5-shared.dir/H5Cmpio.c.o" "CMakeFiles/hdf5-shared.dir/H5Cprefetched.c.o" "CMakeFiles/hdf5-shared.dir/H5Cquery.
             c.o" "CMakeFiles/hdf5-shared.dir/H5Ctag.c.o" "CMakeFiles/hdf5-shared.dir/H5Ctest.c.o" "CMakeFiles/hdf5-shared.dir/H5CS.c.o" "CMakeFiles/hdf5-share
             d.dir/H5CX.c.o" "CMakeFiles/hdf5-shared.dir/H5D.c.o" "CMakeFiles/hdf5-shared.dir/H5Dbtree.c.o" "CMakeFiles/hdf5-shared.dir/H5Dbtree2.c.o" "CMakeFi
             les/hdf5-shared.dir/H5Dchunk.c.o" "CMakeFiles/hdf5-shared.dir/H5Dcompact.c.o" "CMakeFiles/hdf5-shared.dir/H5Dcontig.c.o" "CMakeFiles/hdf5-shared.d
             ir/H5Ddbg.c.o" "CMakeFiles/hdf5-shared.dir/H5Ddeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Dearray.c.o" "CMakeFiles/hdf5-shared.dir/H5Defl.c.o" "CMak
             eFiles/hdf5-shared.dir/H5Dfarray.c.o" "CMakeFiles/hdf5-shared.dir/H5Dfill.c.o" "CMakeFiles/hdf5-shared.dir/H5Dint.c.o" "CMakeFiles/hdf5-shared.dir
             /H5Dio.c.o" "CMakeFiles/hdf5-shared.dir/H5Dlayout.c.o" "CMakeFiles/hdf5-shared.dir/H5Dmpio.c.o" "CMakeFiles/hdf5-shared.dir/H5Dnone.c.o" "CMakeFil
             es/hdf5-shared.dir/H5Doh.c.o" "CMakeFiles/hdf5-shared.dir/H5Dscatgath.c.o" "CMakeFiles/hdf5-shared.dir/H5Dselect.c.o" "CMakeFiles/hdf5-shared.dir/
             H5Dsingle.c.o" "CMakeFiles/hdf5-shared.dir/H5Dtest.c.o" "CMakeFiles/hdf5-shared.dir/H5Dvirtual.c.o" "CMakeFiles/hdf5-shared.dir/H5E.c.o" "CMakeFil
             es/hdf5-shared.dir/H5Edeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Eint.c.o" "CMakeFiles/hdf5-shared.dir/H5EA.c.o" "CMakeFiles/hdf5-shared.dir/H5EAca
             che.c.o" "CMakeFiles/hdf5-shared.dir/H5EAdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5EAdblkpage.c.o" "CMakeFiles/hdf5-shared.dir/H5EAdblock.c.o" "CMake
             Files/hdf5-shared.dir/H5EAhdr.c.o" "CMakeFiles/hdf5-shared.dir/H5EAiblock.c.o" "CMakeFiles/hdf5-shared.dir/H5EAint.c.o" "CMakeFiles/hdf5-shared.di
             r/H5EAsblock.c.o" "CMakeFiles/hdf5-shared.dir/H5EAstat.c.o" "CMakeFiles/hdf5-shared.dir/H5EAtest.c.o" "CMakeFiles/hdf5-shared.dir/H5ES.c.o" "CMake
             Files/hdf5-shared.dir/H5ESevent.c.o" "CMakeFiles/hdf5-shared.dir/H5ESint.c.o" "CMakeFiles/hdf5-shared.dir/H5ESlist.c.o" "CMakeFiles/hdf5-shared.di
             r/H5F.c.o" "CMakeFiles/hdf5-shared.dir/H5Faccum.c.o" "CMakeFiles/hdf5-shared.dir/H5Fcwfs.c.o" "CMakeFiles/hdf5-shared.dir/H5Fdbg.c.o" "CMakeFiles/
             hdf5-shared.dir/H5Fdeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Fefc.c.o" "CMakeFiles/hdf5-shared.dir/H5Ffake.c.o" "CMakeFiles/hdf5-shared.dir/H5Fint
             .c.o" "CMakeFiles/hdf5-shared.dir/H5Fio.c.o" "CMakeFiles/hdf5-shared.dir/H5Fmount.c.o" "CMakeFiles/hdf5-shared.dir/H5Fmpi.c.o" "CMakeFiles/hdf5-sh
             ared.dir/H5Fquery.c.o" "CMakeFiles/hdf5-shared.dir/H5Fsfile.c.o" "CMakeFiles/hdf5-shared.dir/H5Fspace.c.o" "CMakeFiles/hdf5-shared.dir/H5Fsuper.c.
             o" "CMakeFiles/hdf5-shared.dir/H5Fsuper_cache.c.o" "CMakeFiles/hdf5-shared.dir/H5Ftest.c.o" "CMakeFiles/hdf5-shared.dir/H5FA.c.o" "CMakeFiles/hdf5
             -shared.dir/H5FAcache.c.o" "CMakeFiles/hdf5-shared.dir/H5FAdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5FAdblkpage.c.o" "CMakeFiles/hdf5-shared.dir/H5FA
             dblock.c.o" "CMakeFiles/hdf5-shared.dir/H5FAhdr.c.o" "CMakeFiles/hdf5-shared.dir/H5FAint.c.o" "CMakeFiles/hdf5-shared.dir/H5FAstat.c.o" "CMakeFile
             s/hdf5-shared.dir/H5FAtest.c.o" "CMakeFiles/hdf5-shared.dir/H5FD.c.o" "CMakeFiles/hdf5-shared.dir/H5FDcore.c.o" "CMakeFiles/hdf5-shared.dir/H5FDdi
             rect.c.o" "CMakeFiles/hdf5-shared.dir/H5FDfamily.c.o" "CMakeFiles/hdf5-shared.dir/H5FDhdfs.c.o" "CMakeFiles/hdf5-shared.dir/H5FDint.c.o" "CMakeFil
             es/hdf5-shared.dir/H5FDlog.c.o" "CMakeFiles/hdf5-shared.dir/H5FDmirror.c.o" "CMakeFiles/hdf5-shared.dir/H5FDmpi.c.o" "CMakeFiles/hdf5-shared.dir/H
             5FDmpio.c.o" "CMakeFiles/hdf5-shared.dir/H5FDmulti.c.o" "CMakeFiles/hdf5-shared.dir/H5FDonion.c.o" "CMakeFiles/hdf5-shared.dir/H5FDonion_header.c.
             o" "CMakeFiles/hdf5-shared.dir/H5FDonion_history.c.o" "CMakeFiles/hdf5-shared.dir/H5FDonion_index.c.o" "CMakeFiles/hdf5-shared.dir/H5FDperform.c.o
             " "CMakeFiles/hdf5-shared.dir/H5FDros3.c.o" "CMakeFiles/hdf5-shared.dir/H5FDs3comms.c.o" "CMakeFiles/hdf5-shared.dir/H5FDsec2.c.o" "CMakeFiles/hdf
             5-shared.dir/H5FDspace.c.o" "CMakeFiles/hdf5-shared.dir/H5FDsplitter.c.o" "CMakeFiles/hdf5-shared.dir/H5FDstdio.c.o" "CMakeFiles/hdf5-shared.dir/H
             5FDtest.c.o" "CMakeFiles/hdf5-shared.dir/H5FDwindows.c.o" "CMakeFiles/hdf5-shared.dir/H5FL.c.o" "CMakeFiles/hdf5-shared.dir/H5FO.c.o" "CMakeFiles/
             hdf5-shared.dir/H5FS.c.o" "CMakeFiles/hdf5-shared.dir/H5FScache.c.o" "CMakeFiles/hdf5-shared.dir/H5FSdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5FSint.
             c.o" "CMakeFiles/hdf5-shared.dir/H5FSsection.c.o" "CMakeFiles/hdf5-shared.dir/H5FSstat.c.o" "CMakeFiles/hdf5-shared.dir/H5FStest.c.o" "CMakeFiles/
             hdf5-shared.dir/H5G.c.o" "CMakeFiles/hdf5-shared.dir/H5Gbtree2.c.o" "CMakeFiles/hdf5-shared.dir/H5Gcache.c.o" "CMakeFiles/hdf5-shared.dir/H5Gcompa
             ct.c.o" "CMakeFiles/hdf5-shared.dir/H5Gdense.c.o" "CMakeFiles/hdf5-shared.dir/H5Gdeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Gent.c.o" "CMakeFiles/h
             df5-shared.dir/H5Gint.c.o" "CMakeFiles/hdf5-shared.dir/H5Glink.c.o" "CMakeFiles/hdf5-shared.dir/H5Gloc.c.o" "CMakeFiles/hdf5-shared.dir/H5Gname.c.
             o" "CMakeFiles/hdf5-shared.dir/H5Gnode.c.o" "CMakeFiles/hdf5-shared.dir/H5Gobj.c.o" "CMakeFiles/hdf5-shared.dir/H5Goh.c.o" "CMakeFiles/hdf5-shared
             .dir/H5Groot.c.o" "CMakeFiles/hdf5-shared.dir/H5Gstab.c.o" "CMakeFiles/hdf5-shared.dir/H5Gtest.c.o" "CMakeFiles/hdf5-shared.dir/H5Gtraverse.c.o" "
             CMakeFiles/hdf5-shared.dir/H5HF.c.o" "CMakeFiles/hdf5-shared.dir/H5HFbtree2.c.o" "CMakeFiles/hdf5-shared.dir/H5HFcache.c.o" "CMakeFiles/hdf5-share
             d.dir/H5HFdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5HFdblock.c.o" "CMakeFiles/hdf5-shared.dir/H5HFdtable.c.o" "CMakeFiles/hdf5-shared.dir/H5HFhdr.c.o
             " "CMakeFiles/hdf5-shared.dir/H5HFhuge.c.o" "CMakeFiles/hdf5-shared.dir/H5HFiblock.c.o" "CMakeFiles/hdf5-shared.dir/H5HFiter.c.o" "CMakeFiles/hdf5
             -shared.dir/H5HFman.c.o" "CMakeFiles/hdf5-shared.dir/H5HFsection.c.o" "CMakeFiles/hdf5-shared.dir/H5HFspace.c.o" "CMakeFiles/hdf5-shared.dir/H5HFs
             tat.c.o" "CMakeFiles/hdf5-shared.dir/H5HFtest.c.o" "CMakeFiles/hdf5-shared.dir/H5HFtiny.c.o" "CMakeFiles/hdf5-shared.dir/H5HG.c.o" "CMakeFiles/hdf
             5-shared.dir/H5HGcache.c.o" "CMakeFiles/hdf5-shared.dir/H5HGdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5HGquery.c.o" "CMakeFiles/hdf5-shared.dir/H5HL.c
             .o" "CMakeFiles/hdf5-shared.dir/H5HLcache.c.o" "CMakeFiles/hdf5-shared.dir/H5HLdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5HLdblk.c.o" "CMakeFiles/hdf5
             -shared.dir/H5HLint.c.o" "CMakeFiles/hdf5-shared.dir/H5HLprfx.c.o" "CMakeFiles/hdf5-shared.dir/H5I.c.o" "CMakeFiles/hdf5-shared.dir/H5Idbg.c.o" "C
             MakeFiles/hdf5-shared.dir/H5Iint.c.o" "CMakeFiles/hdf5-shared.dir/H5Itest.c.o" "CMakeFiles/hdf5-shared.dir/H5L.c.o" "CMakeFiles/hdf5-shared.dir/H5
             Ldeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Lexternal.c.o" "CMakeFiles/hdf5-shared.dir/H5Lint.c.o" "CMakeFiles/hdf5-shared.dir/H5M.c.o" "CMakeFiles
             /hdf5-shared.dir/H5MF.c.o" "CMakeFiles/hdf5-shared.dir/H5MFaggr.c.o" "CMakeFiles/hdf5-shared.dir/H5MFdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5MFsect
             ion.c.o" "CMakeFiles/hdf5-shared.dir/H5MM.c.o" "CMakeFiles/hdf5-shared.dir/H5O.c.o" "CMakeFiles/hdf5-shared.dir/H5Oainfo.c.o" "CMakeFiles/hdf5-sha
             red.dir/H5Oalloc.c.o" "CMakeFiles/hdf5-shared.dir/H5Oattr.c.o" "CMakeFiles/hdf5-shared.dir/H5Oattribute.c.o" "CMakeFiles/hdf5-shared.dir/H5Obogus.
             c.o" "CMakeFiles/hdf5-shared.dir/H5Obtreek.c.o" "CMakeFiles/hdf5-shared.dir/H5Ocache.c.o" "CMakeFiles/hdf5-shared.dir/H5Ocache_image.c.o" "CMakeFi
             les/hdf5-shared.dir/H5Ochunk.c.o" "CMakeFiles/hdf5-shared.dir/H5Ocont.c.o" "CMakeFiles/hdf5-shared.dir/H5Ocopy.c.o" "CMakeFiles/hdf5-shared.dir/H5
             Ocopy_ref.c.o" "CMakeFiles/hdf5-shared.dir/H5Odbg.c.o" "CMakeFiles/hdf5-shared.dir/H5Odeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Odrvinfo.c.o" "CMa
             keFiles/hdf5-shared.dir/H5Odtype.c.o" "CMakeFiles/hdf5-shared.dir/H5Oefl.c.o" "CMakeFiles/hdf5-shared.dir/H5Ofill.c.o" "CMakeFiles/hdf5-shared.dir
             /H5Oflush.c.o" "CMakeFiles/hdf5-shared.dir/H5Ofsinfo.c.o" "CMakeFiles/hdf5-shared.dir/H5Oginfo.c.o" "CMakeFiles/hdf5-shared.dir/H5Oint.c.o" "CMake
             Files/hdf5-shared.dir/H5Olayout.c.o" "CMakeFiles/hdf5-shared.dir/H5Olinfo.c.o" "CMakeFiles/hdf5-shared.dir/H5Olink.c.o" "CMakeFiles/hdf5-shared.di
             r/H5Omessage.c.o" "CMakeFiles/hdf5-shared.dir/H5Omtime.c.o" "CMakeFiles/hdf5-shared.dir/H5Oname.c.o" "CMakeFiles/hdf5-shared.dir/H5Onull.c.o" "CMa
             keFiles/hdf5-shared.dir/H5Opline.c.o" "CMakeFiles/hdf5-shared.dir/H5Orefcount.c.o" "CMakeFiles/hdf5-shared.dir/H5Osdspace.c.o" "CMakeFiles/hdf5-sh
             ared.dir/H5Oshared.c.o" "CMakeFiles/hdf5-shared.dir/H5Oshmesg.c.o" "CMakeFiles/hdf5-shared.dir/H5Ostab.c.o" "CMakeFiles/hdf5-shared.dir/H5Otest.c.
             o" "CMakeFiles/hdf5-shared.dir/H5Ounknown.c.o" "CMakeFiles/hdf5-shared.dir/H5P.c.o" "CMakeFiles/hdf5-shared.dir/H5Pacpl.c.o" "CMakeFiles/hdf5-shar
             ed.dir/H5Pdapl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pdcpl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pdeprec.c.o" "CMakeFiles/hdf5-shared.dir/H5Pdxpl.c.o" "
             CMakeFiles/hdf5-shared.dir/H5Pencdec.c.o" "CMakeFiles/hdf5-shared.dir/H5Pfapl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pfcpl.c.o" "CMakeFiles/hdf5-share
             d.dir/H5Pfmpl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pgcpl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pint.c.o" "CMakeFiles/hdf5-shared.dir/H5Plapl.c.o" "CMak
             eFiles/hdf5-shared.dir/H5Plcpl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pmapl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pmcpl.c.o" "CMakeFiles/hdf5-shared.dir/
             H5Pocpl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pocpypl.c.o" "CMakeFiles/hdf5-shared.dir/H5Pstrcpl.c.o" "CMakeFiles/hdf5-shared.dir/H5Ptest.c.o" "CMake
             Files/hdf5-shared.dir/H5PB.c.o" "CMakeFiles/hdf5-shared.dir/H5PL.c.o" "CMakeFiles/hdf5-shared.dir/H5PLint.c.o" "CMakeFiles/hdf5-shared.dir/H5PLpat
             h.c.o" "CMakeFiles/hdf5-shared.dir/H5PLplugin_cache.c.o" "CMakeFiles/hdf5-shared.dir/H5R.c.o" "CMakeFiles/hdf5-shared.dir/H5Rdeprec.c.o" "CMakeFil
             es/hdf5-shared.dir/H5Rint.c.o" "CMakeFiles/hdf5-shared.dir/H5UC.c.o" "CMakeFiles/hdf5-shared.dir/H5RS.c.o" "CMakeFiles/hdf5-shared.dir/H5S.c.o" "C
             MakeFiles/hdf5-shared.dir/H5Sall.c.o" "CMakeFiles/hdf5-shared.dir/H5Sdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5Sdeprec.c.o" "CMakeFiles/hdf5-shared.d
             ir/H5Shyper.c.o" "CMakeFiles/hdf5-shared.dir/H5Smpio.c.o" "CMakeFiles/hdf5-shared.dir/H5Snone.c.o" "CMakeFiles/hdf5-shared.dir/H5Spoint.c.o" "CMak
             eFiles/hdf5-shared.dir/H5Sselect.c.o" "CMakeFiles/hdf5-shared.dir/H5Stest.c.o" "CMakeFiles/hdf5-shared.dir/H5SL.c.o" "CMakeFiles/hdf5-shared.dir/H
             5SM.c.o" "CMakeFiles/hdf5-shared.dir/H5SMbtree2.c.o" "CMakeFiles/hdf5-shared.dir/H5SMcache.c.o" "CMakeFiles/hdf5-shared.dir/H5SMmessage.c.o" "CMak
             eFiles/hdf5-shared.dir/H5SMtest.c.o" "CMakeFiles/hdf5-shared.dir/H5T.c.o" "CMakeFiles/hdf5-shared.dir/H5Tarray.c.o" "CMakeFiles/hdf5-shared.dir/H5
             Tbit.c.o" "CMakeFiles/hdf5-shared.dir/H5Tcommit.c.o" "CMakeFiles/hdf5-shared.dir/H5Tcompound.c.o" "CMakeFiles/hdf5-shared.dir/H5Tconv.c.o" "CMakeF
             iles/hdf5-shared.dir/H5Tcset.c.o" "CMakeFiles/hdf5-shared.dir/H5Tdbg.c.o" "CMakeFiles/hdf5-shared.dir/H5Tdeprec.c.o" "CMakeFiles/hdf5-shared.dir/H
             5Tenum.c.o" "CMakeFiles/hdf5-shared.dir/H5Tfields.c.o" "CMakeFiles/hdf5-shared.dir/H5Tfixed.c.o" "CMakeFiles/hdf5-shared.dir/H5Tfloat.c.o" "CMakeF
             iles/hdf5-shared.dir/H5Tinit_float.c.o" "CMakeFiles/hdf5-shared.dir/H5Tnative.c.o" "CMakeFiles/hdf5-shared.dir/H5Toffset.c.o" "CMakeFiles/hdf5-sha
             red.dir/H5Toh.c.o" "CMakeFiles/hdf5-shared.dir/H5Topaque.c.o" "CMakeFiles/hdf5-shared.dir/H5Torder.c.o" "CMakeFiles/hdf5-shared.dir/H5Tpad.c.o" "C
             MakeFiles/hdf5-shared.dir/H5Tprecis.c.o" "CMakeFiles/hdf5-shared.dir/H5Tref.c.o" "CMakeFiles/hdf5-shared.dir/H5Tstrpad.c.o" "CMakeFiles/hdf5-share
             d.dir/H5Tvisit.c.o" "CMakeFiles/hdf5-shared.dir/H5Tvlen.c.o" "CMakeFiles/hdf5-shared.dir/H5TS.c.o" "CMakeFiles/hdf5-shared.dir/H5VL.c.o" "CMakeFil
             es/hdf5-shared.dir/H5VLcallback.c.o" "CMakeFiles/hdf5-shared.dir/H5VLdyn_ops.c.o" "CMakeFiles/hdf5-shared.dir/H5VLint.c.o" "CMakeFiles/hdf5-shared
             .dir/H5VLnative.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_attr.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_blob.c.o" "CMakeFiles/hdf5-shared.dir
             /H5VLnative_dataset.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_datatype.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_file.c.o" "CMakeFiles/hdf5-sh
             ared.dir/H5VLnative_group.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_link.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_introspect.c.o" "CMakeFiles
             /hdf5-shared.dir/H5VLnative_object.c.o" "CMakeFiles/hdf5-shared.dir/H5VLnative_token.c.o" "CMakeFiles/hdf5-shared.dir/H5VLpassthru.c.o" "CMakeFile
             s/hdf5-shared.dir/H5VLtest.c.o" "CMakeFiles/hdf5-shared.dir/H5VM.c.o" "CMakeFiles/hdf5-shared.dir/H5WB.c.o" "CMakeFiles/hdf5-shared.dir/H5Z.c.o" "
             CMakeFiles/hdf5-shared.dir/H5Zdeflate.c.o" "CMakeFiles/hdf5-shared.dir/H5Zfletcher32.c.o" "CMakeFiles/hdf5-shared.dir/H5Znbit.c.o" "CMakeFiles/hdf
             5-shared.dir/H5Zscaleoffset.c.o" "CMakeFiles/hdf5-shared.dir/H5Zshuffle.c.o" "CMakeFiles/hdf5-shared.dir/H5Zszip.c.o" "CMakeFiles/hdf5-shared.dir/
             H5Ztrans.c.o"  -Wl,-rpath,@executable_path/../lib -Wl,-rpath,@executable_path/ -Wl,-rpath,@loader_path/../lib -Wl,-rpath,@loader_path/ -lm -ldl /U
             sers/mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/openmpi-4.1.6-5533xhhvzvbsamx55vyh5xvac3hidqgz/lib/libmpi.dylib /Users
             /mathomp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/zlib-ng-2.1.4-x7qn2eenbtlrvogqwblgubufm3af75lh/lib/libz.dylib /Users/matho
             mp4/spack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/libaec-1.0.6-2burp52r3trdkw7oyfay33yc7accx7gv/lib/libsz.dylib /Users/mathomp4/sp
             ack-mathomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/libaec-1.0.6-2burp52r3trdkw7oyfay33yc7accx7gv/lib/libsz.dylib /Users/mathomp4/spack-ma
             thomp4/opt/spack/darwin-ventura-m1/apple-clang-15.0.0/openmpi-4.1.6-5533xhhvzvbsamx55vyh5xvac3hidqgz/lib/libmpi.dylib
     1736    [ 72%] Built target hdf5-static
     1737    cd /private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-bu
             ild-5jf4yhy/src && /Users/mathomp4/.homebrew/brew/Cellar/cmake/3.27.8/bin/cmake -E cmake_symlink_library ../bin/libhdf5.310.3.0.dylib ../bin/libhd
             f5.310.dylib ../bin/libhdf5.dylib
     1738    make[2]: Leaving directory '/private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvye
             hzyuzte6xzn53wva/spack-build-5jf4yhy'
     1739    [ 72%] Built target hdf5-shared
     1740    make[1]: Leaving directory '/private/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvye
             hzyuzte6xzn53wva/spack-build-5jf4yhy'
  >> 1741    make: *** [Makefile:159: all] Error 2

See build log for details:
  /var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/mathomp4/spack-stage/spack-stage-hdf5-1.14.3-5jf4yhyq6krwqvyehzyuzte6xzn53wva/spack-build-out.txt</pre></details>


### Information on your system

* **Spack:** 0.22.0.dev0 (34d791189da2ba941bb8d8e3b8be202f0d855e97)
* **Python:** 3.11.6
* **Platform:** darwin-ventura-m1
* **Concretizer:** clingo

As for yaml files, here are my compilers, modules, and packages (renamed as .txt so GitHub would allow them to be uploaded:

* [compilers.txt](https://github.com/spack/spack/files/13514102/compilers.txt)
* [modules.txt](https://github.com/spack/spack/files/13514103/modules.txt)
* [packages.txt](https://github.com/spack/spack/files/13514104/packages.txt)

Maintainer Mentions:
@brtnfld @byrnHDF @gheber @hyoklee @lkurz @lrknox 

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/13514126/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/13514127/spack-build-out.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hdf5/package.py
+++ b/var/spack/repos/builtin/packages/hdf5/package.py
@@ -293,9 +293,13 @@ class Hdf5(CMakePackage):
                 cmake_flags.append(self.compiler.cc_pic_flag)
             if spec.satisfies("@1.8.21 %oneapi@2023.0.0"):
                 cmake_flags.append("-Wno-error=int-conversion")
+            if spec.satisfies("%apple-clang@15:"):
+                cmake_flags.append("-Wl,-ld_classic")
         elif name == "cxxflags":
             if spec.satisfies("@:1.8.12+cxx~shared"):
                 cmake_flags.append(self.compiler.cxx_pic_flag)
+            if spec.satisfies("%apple-clang@15:"):
+                cmake_flags.append("-Wl,-ld_classic")
         elif name == "fflags":
             if spec.satisfies("%cce+fortran"):
                 # Cray compiler generates module files with uppercase names by

